### ISSUE TITLE
More obfuscation crap from CDA.pl player

### ISSUE THREAD

**Issue by Bartixxx32** [divadsn/cda-video-extractor#3]
```
function x1(a) {
    a = a.replace(".cda.mp4", "");
    a = a.replace(".2cda.pl", ".cda.pl");
    a = a.replace(".3cda.pl", ".cda.pl");
    return "https://" + a + ".mp4"
};

function x2(a) {
    return decodeURIComponent(a)
};

function x3(a) {
    for (var b = [], e = 0; e < a.length; e++) {
        var f = a.charCodeAt(e);
        b[e] = 33 <= f && 126 >= f ? String.fromCharCode(33 + (f + 14) % 94) : String.fromCharCode(f)
    }
    return x1(b.join(""))
};

function x4(a) {
    return x7(x2(x6(a)))
};

function x5(a) {
    String.fromCharCode(("Z" >= a ? 11 : 344) >= (c = a.charCodeAt(0) + 22) ? c : c - 11);
    a = a.replace("_XDDD", "");
    a = a.replace("_CDA", "");
    a = a.replace("_ADC", "");
    a = a.replace("_CXD", "");
    a = a.replace("_QWE", "");
    a = a.replace("_Q5", "");
    return x4(x6(a))
};

function x6(a) {
    return a.replace(/[a-zA-Z]/g, function(a) {
        return String.fromCharCode(("Z" >= a ? 90 : 122) >= (a = a.charCodeAt(0) + 13) ? a : a - 26)
    })
};

function x7(a) {
    String.fromCharCode(("Z" >= a ? 82 : 132) >= (c = a.charCodeAt(0) + 11) ? c : c - 55);
    return x3(a)
};

function x8(a) {
    return !B(a, "http") && !B(a, ".mp4") && !B(a, "uggcf://")
};
```

**Comment by divadsn** [comment]
Tak, generalnie nieźle kombinują, aczkolwiek głównie dodają teraz nonsens do linku, bo lista tych "żartów" rośnie, więc przydało by się regexem te słowa wyciągnąć jakoś.

**Comment by divadsn** [comment]
PR #2 zawiera niektóre już z nich, ale odrzuciłem z racji na poprawki inne

**Comment by Bartixxx32** [comment]
rozumiem

**Comment by Bartixxx32** [comment]
Ale zauważ że pobieracz.net jakoś to obszedł, nie wiem jak oni czarują ale, do tego dali extractor portable, pobierasz jakby ich source strony, i odpalasz php lokalnie i jakoś działa, ale przez api sie łączy udało mi sie tylko zdecryptować base64, ale ten kod to było jakieś łączenia do api,I co ciekawe używają useragenta ```Mozilla/5.0 (Linux; Tizen 2.3; SmartHub; SMART-TV; SmartTV; U; Maple2012) AppleWebKit/538.1+ (KHTML, like Gecko) TV Safari/538.1+```

**Comment by Bartixxx32** [comment]
I ogólnie sie zastanawiałem czy dało by rade zrobić jakiś driver do chromium.
Że działał by jakoś na zasadzie clienta który otwiera i ładuje strone, po czym dumpuje caly kod html jaki widział bo normalnie możesz wyszukać .mp4 w inspect element. To szyfrowanie działa tylko do skryptów z jakiegoś powodu
![image](https://user-images.githubusercontent.com/16412925/79056051-b12e8580-7c52-11ea-941d-e04fc26e8430.png)

**Comment by Bartixxx32** [comment]
PS. można sie jakoś skontaktować? np discord lub cos innego? może coś by się wymyśliło.

**Comment by divadsn** [comment]
> I ogólnie sie zastanawiałem czy dało by rade zrobić jakiś driver do chromium.

Takie kombinacje wymagały by postawienia środowiska graficznego na serwerze, nawet w minimalnej konfiguracji typu sam czysty Xserver było by to już słabo wydajne, co widać po np. cdax.pl

> Ale zauważ że pobieracz.net jakoś to obszedł, nie wiem jak oni czarują ale, do tego dali extractor portable, pobierasz jakby ich source strony

Pobieracz.net po prostu na bieżąco trzyma listę z tymi wszystkimi słowami które ostatnio CDA.pl na siłe wciska do linków aby zepsuć działanie stron typu jak ten projekt demonstruje.

Myślę że wystarczy jak ogarnie się RegExem wydobycie tej listy słów do usunięcia z skryptu playera.

**Comment by Bartixxx32** [comment]
Przynajmniej na plus że DRM nie ma xD

**Comment by Bartixxx32** [comment]
Tak właściwie dużo jest to roboty aby zaadaptować twój commit co naprawia znowu extractor aby działało to na forku youtube-dl? Bo mam wlasny projekt do pobierania całych folderów z cda i filmów w nich, ale bez youtube-dl leży i kwiczy

**Comment by divadsn** [comment]
> Tak właściwie dużo jest to roboty aby zaadaptować twój commit co naprawia znowu extractor aby działało to na forku youtube-dl? Bo mam wlasny projekt do pobierania całych folderów z cda i filmów w nich, ale bez youtube-dl leży i kwiczy

Był PR robiony do youtube-dl, ale z racji na ciągle zmieniającą się sytuację usunąłem forka na razie.
